### ISSUE TITLE
This plugin does not seem to work with Android projects

### ISSUE THREAD

**Issue by vRallev** [JLLeitschuh/ktlint-gradle#8]
According to the source and my own tests

**Comment by JLLeitschuh** [comment]
That may be true.

I'm not sure how android projects are structured. I basically look for every source set that the `JavaPluginConvention` is aware of, iterate over each and apply this plugin.

https://github.com/JLLeitschuh/ktlint-gradle/blob/master/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt#L43-L78

What problem are you having??

Note, this plugin does **_not_** automatically configure itself so that it's attached to the `check` task. You have to do this manually.

**Comment by vRallev** [comment]
Android is using a different plugin `kotlin-lint`. A few other things would need to be changed as well, e.g. the source set detection.

**Comment by JLLeitschuh** [comment]
I haven't heard of `kotlin-lint` (link??) is this something you are interested in opening a PR to fix?

I don't work in the android ecosystem and I don't really need the feature. If you are interested in seeing it added please feel free to open a PR.

**Comment by vRallev** [comment]
I have no idea why I wrote kotlin-lint. I actually meant `kotlin-android`. Sorry

**Comment by Tapchicoma** [comment]
So I've added ktlint to my project and see that plugin created tasks for android modules that use kotlin as well, but performs checks only on kotlin files in java module.

**Comment by Tapchicoma** [comment]
I think the problem is that on Android java sourcesets are used for kotlin files as well. So this lines:

```kotlin
val kotlinSourceSet: SourceDirectorySet = (it as HasConvention)
        .convention
        .getPluginHelper<KotlinSourceSet>()
        .kotlin
```
doesn't returns any source dirs

**Comment by JLLeitschuh** [comment]
Pull requests that fix this problem will be accepted.

**Comment by JLLeitschuh** [comment]
Closed by #12
